benchmark/array_extract.rb

```diff
@@ -0,0 +1,11 @@
+require 'benchmark'
+
+arrays = Array.new(1000) { (0..10000).to_a }
+
+Benchmark.bmbm do |x|
+  x.report('Array#extract!') do
+    arrays.each do |numbers|
+      _odd_numbers = numbers.extract! { |number| number.odd? }
+    end
```

Comment:
This is not correct, as `bmbm` will run this block twice, but the nested arrays will be mutated.
That's also why the second set of results on https://bugs.ruby-lang.org/issues/15831 is quite different from the first run.

Reply:
Thanks. I just fixed it. Also, I removed benchmarks from the [description](https://bugs.ruby-lang.org/issues/15831), since they are now located in `benchmark/`.